### ISSUE TITLE
Simple Example for generating krigged grid of sparse data

### ISSUE THREAD

**Issue by hal98x** [capaulson/pyKriging#7]
Not really an issue, Just can't follow the examples and see how to implement this library for my own simple task.

Do you have an example similar to the ones at this website http://hackersome.com/p/bsmurphy/PyKrige.

It could be that I am missing the point of the library.

Sincerely,

Michael Wallace, P.Geo., PGp

**Comment by capaulson** [comment]
Hi Michael, 

Sorry that the examples aren't clear. I've included a bare-bones example below that attempts to show how to use pyKriging. The X array represents an array of vectors where observations are made, while the y array holds the corresponding scalar for each observed point. Once a Kriging model is initialized with the observed data, you need to train the model (k.train()). Training the model establishes the hyperparameters for the model. Once trained, you can query the model (k.predict()) or plot the result.

Does this help? 

Best, 
Chris 

```
import numpy as np 
from pyKriging import kriging
X = np.array([[ 0.75,  0.15],
       [ 0.55,  0.95],
       [ 0.15,  0.25],
       [ 0.25,  0.85],
       [ 0.05,  0.65],
       [ 0.35,  0.45],
       [ 0.45,  0.05],
       [ 0.85,  0.75],
       [ 0.65,  0.55],
       [ 0.95,  0.35]])

y = np.array( [0.865, 0.325, 0.085, 0.065 ,0.125, 0.325, 0.925, 1.145, 1.325, 0.9025] )

k = kriging(X, y)
k.train()
print k.predict([.25,.25])
k.plot()
```

**Comment by hal98x** [comment]
Thanks for getting back too me, I have been using pykrige for awhile and there are some performance issues with the newest version, so now I am back testing your library. The example ran perfectly and so I tried extending it to my data. I followed the example with my 2225 measurements and I get the following exception 

  File "C:\Users\mwallace\AppData\Local\Continuum\Anaconda2\lib\site-packages\pyKriging\krige.py", line 145, in updateModel
    raise Exception("bad params")

Exception: bad params

Also do you have any doc on the inputs to k.plot and kriging?

**Comment by capaulson** [comment]
Hmmm... If I had to guess (without seeing the data), you have two identical points or two points that are very close to each other. This causes issues in matrix calculations. Could you share data?

**Comment by hal98x** [comment]
I removed all the points that where less than a meter apart from each other and then executed the code again. No luck still get bad params error. Attached is a csv with the 3 columns of data. 


[pyKrigging_test.txt](https://github.com/capaulson/pyKriging/files/256099/pyKrigging_test.txt)

# setup grid
xmin=np.min(x)
xmax=np.max(x)
ymin=np.min(y)
ymax=np.max(y)

dx=(xmax-xmin)
dy=(ymax-ymin)
gridy = np.arange(0, dy, dy/100.0)
gridx = np.arange(0, dx, dx/100.0)

grid=np.vstack((gridx,gridy)).T

# from example in this thread
k = kriging(pos, z)
k.train()
znew=k.predict(grid)
k.plot()

**Comment by capaulson** [comment]
Michael, 

Thank you for sharing the data. This issue was fairly obvious. I believe it should be addressed by this commit:

03e02c4677814e9d9dd04da90a8d6c55fa6e020f

I'm running the model right now, but it is taking a while. I'll report back once I get it trained to verify that it worked. 

Best, 
Chris

**Comment by capaulson** [comment]
Michael, 

I wanted to follow up with results. To save time on training, I reduced the point count to 500 randomly selected points. The included plot shows the results. The upper left is a contour plot of the prediction, upper right is the predicted MSE, the lower plot is a 3D plot of the prediction model.
![pykriging_issue_7](https://cloud.githubusercontent.com/assets/210252/15154312/b05a49a8-1691-11e6-8f71-889b0da20a60.png)

**Comment by capaulson** [comment]
This is the Kriging model with all the data:
![pykriging_issue_7](https://cloud.githubusercontent.com/assets/210252/15161114/c0e8cd38-16b1-11e6-93db-db2f4c6d0de2.png)

**Comment by hal98x** [comment]
Thanks this is great and looks really good and its in the same color scale I use daily. 

I figured out how to update my system and then I tried running the code. Performance on my computer is slow even when I decimate the data. I was unable to get past the training step (no error) just killed the process because it had been running for too long (over an hour for the big set). It is interesting though to compare the decimated to undecimated data. It says I don't need that dense of data to get a decent interpretation. 

Your observation about overlapping points was also helpful because it gave me a way to filter all the data for overlap instead of just clusters. I think I need to use a compiled library like gslib, surfer win32com events, qgis or sage possibly. 

I think I will use pyKrigging for another application where there is less and sparser data. Thanks capaulson for the help and being responsive.

Sincerely,


Michael Wallace

**Comment by mygaurav** [comment]
Hi capaulson, I am getting bad params error due to positive definiteness with same set of data in one run, and not in another run. Is it because of the random seed used for differential evolution optimisation ? I am stuck on this, tried pulling recent releases, even that did not help. How to ensure positive definiteness of correlation matrix

**Comment by WhitCrow** [comment]
why not have more detailed documentation!
